spec/models/king_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe King, type: :model do
+	let(:king) {create :king}
+
+  describe "King's valid_move" do
+  	it "should be inside destination, blocked" do
+  		expect(king.valid_move?(1, 6)).to eq(false)
+  		expect(king.valid_move?(1, 4)).to eq(false)
+  		expect(king.valid_move?(2, 5)).to eq(false)
+  	end
```

Comment:
Could you add another `expect` statement here to make sure that moves down to a lower rank are also properly categorized as invalid when they are blocked? You can use the black king to test this.